### PR TITLE
scm: add `title` for resources

### PR DESCRIPTION
**Pull request by vince-fugnitto**
#### What it does

Fixes: https://github.com/eclipse-theia/theia/issues/9020.

The commit updates the `scm` view to add `tooltips` or `titles` to the elements in the view so end-users can see the full path of resources and folders.

https://user-images.githubusercontent.com/40359487/125822026-004c43ee-e9e1-4ae0-827e-6d52d029806e.mp4

#### How to test

1. start the application with a workspace folder under git version control (ex: theia)
2. perform changes to the workspace (at the root, nested folders, etc.)
3. open the `scm` view
4. verify that changes in the view display the path when hovering over the entries - confirm with tree and list modes

#### Review checklist

- [x] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

Signed-off-by: vince-fugnitto <vincent.fugnitto@ericsson.com>

### CODE REVIEW

**Comment by colin-grant-work** [comment]
This does indeed add titles to the tree nodes in the SCM tree. I notice, though, that in VSCode the title includes the source control status (modified, new, deleted, etc.) of the file separated from the path by a dot. Would you like to do the same here, or put that off for another PR?

**Comment by vince-fugnitto** [comment]
> This does indeed add titles to the tree nodes in the SCM tree. I notice, though, that in VSCode the title includes the source control status (modified, new, deleted, etc.) of the file separated from the path by a dot. Would you like to do the same here, or put that off for another PR?

Nice suggestion :) I made the update to append the decoration tooltip (ex: `modified`, `deleted`, ...)  to the tooltip along with the uri.

File: packages/scm/src/browser/scm-tree-widget.tsx [hunk 1]
```diff
@@ -526,15 +526,20 @@ export class ScmResourceComponent extends ScmElement<ScmResourceComponent.Props>
         const icon = labelProvider.getIcon(resourceUri);
         const color = decoration && decoration.colorId ? `var(${colors.toCssVariableName(decoration.colorId)})` : '';
         const letter = decoration && decoration.letter || '';
-        const tooltip = decoration && decoration.tooltip || '';
+        const tooltip: string = decoration && decoration.tooltip || '';
```
Comment (msujew) [inline packages/scm/src/browser/scm-tree-widget.tsx:529]: Nitpick: As far as I can tell, this `string` typing is not necessary. TypeScript should be able to infer the type correctly as `string`. Tested with TS 4.3.2:

image

**Review by msujew — APPROVED** [review]
I can confirm that this enables users to see the full path of the changed file and its scm status. This aligns the behavior to vscode 👍

File: packages/scm/src/browser/scm-tree-widget.tsx (hunk 1 above)
Comment (vince-fugnitto) [reply packages/scm/src/browser/scm-tree-widget.tsx]: For some reason if was not inferring it for me (`any`), but I can confirm it works now when restarting the language server :+1:

**Comment by vince-fugnitto** [comment]
@colin-grant-work I've updated the pull-request following your suggestion to also included the scm status as part of the tooltip.

**Review by colin-grant-work — APPROVED** [review]

**Pull request merged**